tests/framework/pkg/session/session.go

```diff
@@ -34,13 +39,13 @@ func (ts *Session) RegisterCleanupFunc(f CleanupFunc) {
 // Cleanup this method will call all registered cleanup functions in order and close the test session.
 func (ts *Session) Cleanup() {
 	ts.open = false
-
 	for i := len(ts.cleanupQueue) - 1; i >= 0; i-- {
 		err := ts.cleanupQueue[i]()
 		if err != nil {
 			ts.testingT.Logf("error calling cleanup function: %v", err)
 		}
 	}
+	ts.cleanupQueue = []CleanupFunc{}
```

Comment:
cleanup functions should be idempotent we don't want to clear them so we can call them again.

Reply:
I was seeing issues when cleaning up resources using the same functions more than once. I figured If it cleanups once it's not necessary to do it again.

Reply:
if that is the case we should be fixing the cleanup functions.

Reply:
thinking about how we use sessions more this makes sense.